com.ca.lsp.cobol/lsp-service-cobol/src/test/java/com/ca/lsp/cobol/FileSystemConfiguration.java

```diff
@@ -85,36 +92,35 @@
    * the filesystem service capabilities using the physical filesystem
    */
   @Before
-  public void buildFS() {
-    intializeSettings();
+  public void buildTempFilesystem() {
+    intializeSettingsWithSomeContent();
+
     createWorkspaceFolderStructure();
-    createCopybookStructure();
-    createDependencyFileStructure();
+    createCopybookFolders();
+    createDependencyFolder();
 
-    // populate copybook folder and dependency file with some content
     createCopybookFiles();
+    createNestedCopybooks();
     createDependencyFile();
   }
 
   @After
-  public void cleanupTempFolder() {
+  public void cleanupTempFilesystem() {
     try {
-      Files.walk(getWorkspaceFolderPath())
-          .sorted(Comparator.reverseOrder())
-          .map(Path::toFile)
-          .forEach(File::delete);
+      deleteFolders();
     } catch (IOException e) {
       log.error(e.getMessage());
     }
   }
 
-  /**
-   * This method define the steps necessary to emulate the json settings provided by the user in the
-   * settings.json
-   */
-  private void intializeSettings() {
-    // SettingsProvider settingsProvider = new SettingsProvider();
+  private void deleteFolders() throws IOException {
+    Files.walk(getWorkspaceFolderPath())
+        .sorted(Comparator.reverseOrder())
```

Comment:
Why do you need to sort it?

Reply:
Because I need to remove folders and item recursively.
So if I try to remove a non empty folder I got exception. So I ordered to have the files first and then the folder (that will be removed easily since there isn't any file inside of it).
Here the [reference I used in the past](https://www.baeldung.com/java-delete-directory#conclusion-1).

Reply:
Nice trick, I didn't know about that